tests/attr/test_class_summarizer.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+import torch
+
+from captum.attr import ClassSummarizer, CommonStats
+
+from .helpers.utils import BaseTest
+
+
+class Test(BaseTest):
+    def class_test(self, data, classes, x_sizes):
+        summarizer = ClassSummarizer(stats=CommonStats())
+        for x, y in data:
+            summarizer.update(x, y)
+
+        summ = summarizer.summary
+        self.assertIsNotNone(summ)
+        self.assertIsInstance(summ, list)
+        for s, size in zip(summ, x_sizes):
+            self.assertIsInstance(s, dict)
+            for key in s:
+                self.assertEqual(s[key].size(), size)
+
+        self.assertIsNotNone(summarizer.class_summaries)
+        all_classes = torch.zeros(len(classes))
+        class_summaries = summarizer.class_summaries
+        all_keys = set(class_summaries.keys())
+        for i, clazz in enumerate(classes):
+            self.assertTrue(clazz in class_summaries)
+            all_keys.remove(clazz)
+            all_classes[i] = 1
+
+            summ = class_summaries[clazz]
+            self.assertIsNotNone(summ)
+            self.assertIsInstance(summ, list)
+
+            for s, size in zip(summ, x_sizes):
+                self.assertIsInstance(s, dict)
+                for key in s:
+                    self.assertEqual(s[key].size(), size)
+
+        self.assertEqual(len(all_keys), 0)
+        self.assertEqual(all_classes.sum(), len(classes))
+
+    def test_classes(self):
+        sizes_to_test = [
+            ((1,),),
+            ((3, 2, 10, 3), (1,)),
+            ((20,)),
+        ]
+        classeses = [list(range(100)), ["%d" % i for i in range(100)]]
+        for batch_size in [None, 1, 4]:
+            for sizes, classes in zip(sizes_to_test, classeses):
```

Comment:
It looks like sizes_to_test and classeses aren't the same length, so testing with size (20,) is skipped, did you intend to add another element to classeses?

nit: Also maybe a name instead of classeses (e.g. class_lists_to_test) might be easier to read?

Reply:
Nice catch. Will do the suggested nit as well 👍